roles/repository_db/handlers/main.yml

```diff
@@ -7,6 +7,13 @@
   command: "/usr/local/bin/initialize-repository-db {{ archive_db_host }} {{ archive_db_port }} {{ archive_db_name }} {{ archive_db_user }}"
 
 - name: migrate repository database
+  environment:
+    # If PGHOST is not set, psql tries to look for cluster "main" which doesn't
+    # work:
```

Comment:
What conditions do you find 'main' missing? Installing postgresql on debain creates such a cluster …

Reply:
I was just comparing the difference between the psql on an existing server vs the new one and found this problem.  The old one didn't have cluster set.  I don't know if that's the problem or if it's a problem somewhere else.

On the older server:

```
$ PGCLUSTER=9.4/main psql -U cnxarchive cnxarchive
Error: Cluster 9.4 main does not exist
```

On the newer server:

```
$ PGCLUSTER=9.4/main psql -U cnxarchive cnxarchive
Error: Invalid data directory
```

Reply:
Are these local servers to you, or VMs I have access to? I've never seen this behavior.

Reply:
The new server is local to me.  I just created a new Ubuntu 16.04 server and deployed to it.

This is on qa:

```
karen@qa00:~$ PGCLUSTER=9.4/main psql -U rhaptos repository
Error: Cluster 9.4 main does not exist
```

Reply:
```
reedstrm@qa00:~$ pg_lsclusters 
Ver Cluster Port Status Owner Data directory Log file
reedstrm@qa00:~$ sudo pg_lsclusters 
Ver Cluster Port Status Owner    Data directory               Log file
9.4 main    5432 online postgres /var/lib/postgresql/9.4/main /var/log/postgresql/postgresql-9.4-main.log
reedstrm@qa00:~$ PGCLUSTER=9.4/main psql -U rhaptos repository
Error: Cluster 9.4 main does not exist
reedstrm@qa00:~$ sudo PGCLUSTER=9.4/main psql -U rhaptos repository
psql (9.4.12)
Type "help" for help.

repository=>

Reply:
Checking changelog for postgresql-common, now.

Reply:
Did we run into this?

Reply:
I haven't tried `sudo`

Reply:
This is what we use to create databases:

https://github.com/Connexions/cnx-deploy/blob/4eb972f1ba3772c75aa9dc337492ec116973095d/tasks/create_db.yml

I haven't looked at what ansible does.